include/enrico/coupled_driver.h

```diff
@@ -23,7 +23,7 @@ class CoupledDriver {
   //!
   //! \param comm The MPI communicator used for the coupled driver
   //! \param node XML node containing settings
-  explicit CoupledDriver(MPI_Comm comm, pugi::xml_node node);
+  explicit CoupledDriver(Comm comm, pugi::xml_node node);
```

Comment:
I think perhaps the drivers should still accept an `MPI_Comm` because that is what the "user" already has. It seems a little awkward that they have to create a `Comm` instance first before being able to create the driver. An alternative would be to still take a `Comm` but allow implicit conversion of `MPI_Comm` to `Comm` which would effectively allow the user to just pass an `MPI_Comm`.

Reply:
I was in two minds about this, too, and I agree.